### ISSUE TITLE
Documentation Error Entrez

### ISSUE THREAD

**Issue by qarqar** [biopython/biopython#1003]
Summary :The present documentation for Entrez access has a bad example which will infinite loop and rapidly and repeatedly  access the Entrez database and may lead to the user to being banned from Entrez.

At http://biopython.org/DIST/docs/tutorial/Tutorial.html#sec:entrez-webenv section **9.15.1  Searching for and downloading sequences using the history**

replace 
```python
from Bio import Entrez
import time
try:
    from urllib.error import HTTPError  # for Python 3
except ImportError:
    from urllib2 import HTTPError  # for Python 2
batch_size = 3
out_handle = open("orchid_rpl16.fasta", "w")
for start in range(0, count, batch_size):
    end = min(count, start+batch_size)
    print("Going to download record %i to %i" % (start+1, end))
    attempt = 1
    while attempt <= 3:
        try:
            fetch_handle = Entrez.efetch(db="nucleotide", rettype="fasta", retmode="text",
                                         retstart=start, retmax=batch_size,
                                         webenv=webenv, query_key=query_key)
        except HTTPError as err:
            if 500 <= err.code <= 599:
                print("Received error from server %s" % err)
                print("Attempt %i of 3" % attempt)
                attempt += 1
                time.sleep(15)
            else:
                raise
    data = fetch_handle.read()
    fetch_handle.close()
    out_handle.write(data)
out_handle.close()
```
++++++++++++++++++++++++++++++++++++++++++++++++++++++

**with**
Example2.py  

[Example2.py.tar.gz](https://github.com/biopython/biopython/files/606623/Example2.py.tar.gz)

----------------------------------------------------------------------------------------------------------
which uses the connectionMade variable to escape out of the infinite loop at  line 
```python
fetch_handle=Entrez.efetch(db="nucleotide",rettype="fasta",retmode="text",
                                            retstart=start, retmax=batch_size,
                                            webenv=webenv,query_key=query_key)
``` 

in the present example

**Comment by peterjc** [comment]
What is the infinite loop? i.e. What conditions trigger it?

Looking at this I would be happier if the ``attempt += 1`` line was moved above the try/except, to ensure it is always incremented.

I can't see the gzip file from the machine I am currently using, what is example 2? Could you just post it here in triple-back-ticks or as a GitHub Gist?

**Comment by peterjc** [comment]
With fresh eyes I can see what you mean - this should be fixed now: 395f08b59efaa3d66ffe9da2a6ec070b559e6ad2

**Comment by peterjc** [comment]
I've updated the online copy of the Tutorial with an interim build of the HTML and PDF files: https://github.com/biopython/DIST/commit/ecad33bd233a245a687d09507a874713fbb001d5

Thank you for reporting this - you're right if someone was trying this out and left it running it could easily annoy the NCBI.
